session/txn.go

```diff
@@ -261,9 +289,22 @@ func (txn *TxnState) Commit(ctx context.Context) error {
 // Rollback overrides the Transaction interface.
 func (txn *TxnState) Rollback() error {
 	defer txn.reset()
+	txn.State = TxnRollingBack
 	return txn.Transaction.Rollback()
 }
 
+// LockKeys tries to lock the keys. Will block until all keys are locked successfully or an error occurs.
+func (txn *TxnState) LockKeys(ctx context.Context, lockCtx *kv.LockCtx, keys ...kv.Key) error {
```

Comment:
Where's this function used?

Reply:
This just overrides the `LockKeys` in `kv.Transaction`, which IMO will be called when TiDB want to lock some keys in KV store.

Reply:
I'm afraid it's not a very good idea to use overriding in such a large golang project which may probably confuse people. But it might be ok if there are unit tests to ensure the correct version is invoked.